minerl/herobraine/hero/handlers/agent/observations/equipped_item.py

```diff
@@ -132,33 +130,45 @@ def __init__(self, keys: List[str], items: list, _default: str, _other: str):
 
     def to_string(self):
         return 'type'
+        # TODO(shwang): Maybe rename this to 'item_id'?
+        #   This field can contain more than just the type now -- also it can contain
+        #   the metadata. (For example, it can be "sandstone" or "sandstone#2").
 
     def from_hero(self, obs_dict):
         try:
             head = obs_dict['equipped_items']
             for key in self._keys:
                 head = head[key]
-            item = head['type']
-            return (self._other if item not in self._items else item)
+            item_type = head['type']
+            metadata = head['metadata']
+            item_id = util.get_unique_matching_item_list_id(self._items, item_type, metadata)
+            if item_id is None:
+                return self._other
+            else:
+                return item_id
         except KeyError:
             return self._default
 
-    def from_universal(self, obs):
+    def from_universal(self, obs) -> str:
         try:
             if self._keys[0] == 'mainhand' and len(self._keys) == 1:
                 offset = -9
                 hotbar_index = obs['hotbar']
                 if obs['slots']['gui']['type'] == 'class net.minecraft.inventory.ContainerPlayer':
                     offset -= 1
 
-                item_name = (
-                    obs['slots']['gui']['slots'][offset + hotbar_index]['name'].split("minecraft:")[-1])
-                if not item_name in self._items:
-                    raise ValueError()
-                if item_name == 'air':
-                    raise KeyError()
-
-                return item_name
+                equip_slot = obs['slots']['gui']['slots'][offset + hotbar_index]
+                item_type = mc.strip_item_prefix(equip_slot['name'])
```

Comment:
TODO to talk about the registry parsing thing provided by MC with this code that parses the items all pretty